File: Day14/Problem_41.cpp
#include<bits/stdc++.h>
using namespace std;


int removeduplicatesfromsortedarray(vector<int>& nums, int n){
    int i=0;
    for(int j= 1;j<n;j++){
        if(nums[i] != nums[j]){
            nums[i+1] = nums[j];
            i = i+1;
        }
    }
    return i+1;
}

int main(){
    int n;
    cin>>n;
    vector<int>nums(n);
    for(int i=0;i<n;i++){
        cin>>nums[i];
    }
    
    int ans = removeduplicatesfromsortedarray(nums, n);
    
    for(int i=0;i<ans;i++){
        cout<<nums[i]<<" ";
    }
    return 0;
}

File: Day14/Problem_41.py
def removeduplicatesfromsortedarray(nums, n):
    i = 0

    for j in range(1, n):
        if nums[i] != nums[j]:
            nums[i + 1] = nums[j]
            i = i + 1

    return i + 1


if __name__ == "__main__":
    n = int(input())

    nums = list(map(int, input().split()))

    ans = removeduplicatesfromsortedarray(nums, n)

    for i in range(ans):
        print(nums[i], end=" ")